core/src/databases_store/store.rs

```diff
@@ -42,7 +42,7 @@ pub struct PostgresDatabasesStore {
 impl PostgresDatabasesStore {
     pub async fn new(db_uri: &str) -> Result<Self> {
         let manager = PostgresConnectionManager::new_from_stringlike(db_uri, NoTls)?;
-        let pool = Pool::builder().max_size(16).build(manager).await?;
+        let pool = Pool::builder().max_size(128).build(manager).await?;
```

Comment:
Can we find a middle ground between 16 and 128, or explain why such a bump? YPAHT but the max number of connections on this instance is 500 (metrics explorer [link](https://console.cloud.google.com/monitoring/metrics-explorer;startTime=2024-10-28T08:03:59Z;endTime=2024-10-29T08:03:59Z?pageState=%7B%22xyChart%22:%7B%22constantLines%22:%5B%7B%22label%22:%22%22,%22targetAxis%22:%22Y1%22,%22value%22:500%7D%5D,%22dataSets%22:%5B%7B%22legendTemplate%22:%22Active%20connections%22,%22plotType%22:%22LINE%22,%22targetAxis%22:%22Y1%22,%22timeSeriesFilter%22:%7B%22aggregations%22:%5B%7B%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22metadata.system_labels.%5C%22name%5C%22%22%5D,%22perSeriesAligner%22:%22ALIGN_MAX%22%7D%5D,%22apiSource%22:%22DEFAULT_CLOUD%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22filter%22:%22metric.type%3D%5C%22cloudsql.googleapis.com%2Fdatabase%2Fpostgresql%2Fnum_backends%5C%22%20resource.type%3D%5C%22cloudsql_database%5C%22%20resource.label.%5C%22database_id%5C%22%3D%5C%22or1g1n-186209:dust-databases-store-db%5C%22%22,%22groupByFields%22:%5B%22metadata.system_labels.%5C%22name%5C%22%22%5D,%22minAlignmentPeriod%22:%2260s%22,%22perSeriesAligner%22:%22ALIGN_MAX%22%7D%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22y1Axis%22:%7B%22label%22:%22%22,%22scale%22:%22LINEAR%22%7D%7D%7D&project=or1g1n-186209&hl=en)), and we are currently running 3 `core` pods which means with this setting we can't add one more pod.

Even though connections are created lazily as they're needed, we should always ensure that max_pool_size * number of pods < max number of connections on the DB.